packages/nextjs/src/server/utils/getAuth.ts

```diff
@@ -0,0 +1,50 @@
+import { createGetToken } from '@clerk/backend-core';
+import { AuthStatus } from '@clerk/backend-core';
+import { NextRequest } from 'next/server';
+
+import { AuthStatusMapping, SessionsApi } from '../types';
+
+export function createGetAuth(sessionsApi: SessionsApi) {
+  return function (req: NextRequest) {
+    // When the __clerk query param is set, we trust that middleware already ran
+    // Then, we don't have to run any re-verify the JWT here, we
+    // can just strip out the claims manually.
+
+    const clerkParam = req.nextUrl.searchParams.get('__clerk');
+
+    if (!clerkParam) {
+      throw 'You need to use "runClerkMiddleware" on your Next.js middleware file.';
```

Comment:
```suggestion
      throw 'You need to use "runClerkMiddleware" on your Next.js middleware.js file.';
```

Let's also add a link to Next.js middleware docs.

Reply:
Added